packages/@aws-cdk/aws-codeguruprofiler/lib/profiling-group.ts

```diff
@@ -0,0 +1,86 @@
+import { Grant, IGrantable } from '@aws-cdk/aws-iam';
+import { Construct, Resource } from '@aws-cdk/core';
+import { CfnProfilingGroup } from './codeguruprofiler.generated';
+
+/**
+ * Properties for creating a new Profiling Group.
+ */
+export interface ProfilingGroupProps {
+
+  /**
+   * A name for the profiling group.
+   */
+  readonly profilingGroupName: string;
```

Comment:
Can this be omitted? Can we produce a sensible default here?

Reply:
i've changed the approach to the following, not an expert by any means, let me know your thoughts.

```ts
physicalName: props.profilingGroupName || Lazy.stringValue({ produce: () => this.node.uniqueId })
```